operational/flexera/msp/customer_usage_insights_report/flexera_msp_customer_usage_insights_report.pt

```diff
@@ -0,0 +1,1425 @@
+
+  var result = {
+    start_at: start_at.toISOString().substring(0, 7),
+    end_at: end_at.toISOString().substring(0, 7)
+  }
+EOS
+end
+
+# Get applied policy metadata for use later
+datasource "ds_applied_policy" do
+  request do
+    auth $auth_flexera
+    host val($ds_flexera_api_hosts, 'flexera')
+    path join(["/policy/v1/orgs/", rs_org_id, "/projects/", rs_project_id, "/applied-policies/", policy_id])
+  end
+end
+
+# Template
+datasource "ds_applied_policy_template" do
+  request do
+    auth $auth_flexera
+    host val($ds_flexera_api_hosts, 'flexera')
+    path join(["/policy/v1/orgs/", rs_org_id, "/projects/", rs_project_id, "/policy-templates/", jq($ds_applied_policy, ".policyTemplate.id")])
+    query "view", "source"
+  end
+end
+
+datasource "ds_applied_policy_template_hash" do
+  run_script $js_applied_policy_template_hash, $ds_applied_policy_template
+end
+
+script "js_applied_policy_template_hash", type: "javascript" do
+  parameters "ds_applied_policy_template"
+  result "result"
+  code <<-EOS
+  // TODO string "secret-key" can be replaced with a longer key, which is retrieved from an env only accessible by policy engine or provided per customer to discourage hardcoding and being able to spoof the resulting hash
+  var result = hashHmac("sha256", JSON.stringify(ds_applied_policy_template), "secret-key")
+  EOS
+end
+
+# Get organizations
+datasource "ds_organizations" do
+  request do
+    auth $auth_flexera
+    host val($ds_flexera_api_hosts, 'flexera')
+    path join(["/msp/v1/orgs/", rs_org_id, "/customers"])
+    ignore_status 403 #Ignore Status 403 so we can gracefully handle that kind of error
+  end
+  result do
+    encoding "json"
+    collect jmes_path(response, "[*]") do
+      field "id", jmes_path(col_item, "id")
+      field "name", jmes_path(col_item, "name")
+    end
+  end
+end
+
+datasource "ds_parent_org_billing_centers" do
+  request do
+    auth $auth_flexera
+    host rs_optima_host
+    path join(["/analytics/orgs/", rs_org_id, "/billing_centers"])
+    header "Api-Version", "1.0"
+    header "User-Agent", "RS Policies"
+    ignore_status [403]
+  end
+end
+
+# Get parent org dimensions
+# https://api.optima.flexeraeng.com/bill-analysis/orgs/36616/costs/dimensions
+datasource "ds_parent_org_rbds" do
+  request do
+    auth $auth_flexera
+    host val($ds_flexera_api_hosts, 'flexera')
+    path join(["/finops-customizations/v1/orgs/", rs_org_id, "/rule-based-dimensions"])
+  end
+end
+
+datasource "ds_request_parent_org_costs_not_allocated" do
+  run_script $js_request_parent_org_costs_not_allocated, $ds_parent_org_rbds, $ds_parent_org_billing_centers, $param_contractual_exclusion, rs_org_id, rs_org_name
+end
+
+script "js_request_parent_org_costs_not_allocated", type: "javascript" do
+  parameters "ds_parent_org_rbds", "ds_parent_org_billing_centers", "param_contractual_exclusion", "rs_org_id", "rs_org_name"
+  result "result"
+  code <<-EOS
+  // initial filters for all requests
+  var f = null;
```

Comment:
Avoid single letter variable names. Variable names should themselves indicate something about what they are used for.

(Also in the style guide that is WIP)